src/cargo/ops/registry/publish.rs

```diff
@@ -236,15 +238,52 @@ pub fn publish(ws: &Workspace<'_>, opts: &PublishOpts<'_>) -> CargoResult<()> {
                 )?));
             }
 
-            transmit(
+            // Prepare workspace context for error message
+            let workspace_context = if original_packages.len() > 1 {
+                let mut remaining: Vec<_> = original_packages
+                    .iter()
+                    .filter(|id| **id != pkg_id)
+                    .map(|id| {
+                        let pkg = &pkg_dep_graph.packages[&id].0;
+                        format!("{} v{}", pkg.name(), pkg.version())
+                    })
+                    .collect();
+                // Also include any packages that are still waiting for confirmation
+                for id in to_confirm.iter().filter(|id| **id != pkg_id) {
+                    let pkg = &pkg_dep_graph.packages[&id].0;
+                    let entry = format!("{} v{}", pkg.name(), pkg.version());
+                    if !remaining.contains(&entry) {
+                        remaining.push(entry);
+                    }
+                }
```

Comment:
Why do we include these?  They have been published, we just don't know if the server has fully finished processing it yet or not.

As is, the current message is misleading.  It says a package hasn't been published yet.  As a user, that means I need to go in and publish it but then I'll get an error saying its published.